pkgs/games/etlegacy/default.nix

```diff
@@ -0,0 +1,127 @@
+, cmake
+, git
+, glew
+, SDL2
+, zlib
+, minizip
+, libjpeg
+, curl
+, lua5_4
+, libogg
+, libtheora
+, freetype
+, libpng
+, sqlite
+, openal
+, unzip
+, cjson
+}:
+
+let
+  version = "2.81.1";
+  pkgname = "etlegacy";
+  lua = lua5_4;
+  mirror = "https://mirror.etlegacy.com";
+  fetchAsset = {asset, sha256}: fetchurl
+    { url = mirror + "/etmain/" + asset;
+    inherit sha256;
+    };
+  pak0 = fetchAsset
+    { asset = "pak0.pk3";
+    sha256 = "712966b20e06523fe81419516500e499c86b2b4fec823856ddbd333fcb3d26e5";
+    };
+  pak1 = fetchAsset
+    { asset = "pak1.pk3";
+    sha256 = "5610fd749024405b4425a7ce6397e58187b941d22092ef11d4844b427df53e5d";
+    };
+  pak2 = fetchAsset
+    { asset = "pak2.pk3";
+    sha256 = "a48ab749a1a12ab4d9137286b1f23d642c29da59845b2bafc8f64e052cf06f3e";
+    };
+  fakeGit = writeScriptBin "git" ''
+    #! ${stdenv.shell} -e
+    if [ "$1" = "describe" ]; then
+      if [ -r VERSION.txt ]; then
+        . <(sed 's/^\(VERSION_[A-Z]\+\) \(.*\)/\1=\2/' VERSION.txt)
+        VERSION="$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH"
+      fi
+      if [ "$VERSION" != "" ]; then
+        echo $VERSION
+      else
+        echo "Unable to determine version"
+        exit 1
+      fi
+    fi
+  '';
+  gamedir = stdenv.mkDerivation rec {
+    pname = pkgname;
+    inherit version;
+
+    src = fetchgit {
+      url = "https://github.com/etlegacy/etlegacy.git";
+      rev = "refs/tags/v" + version;
+      sha256 = "sha256-CGXtc51vaId/SHbD34ZeT0gPsrl7p2DEw/Kp+GBZIaA=";  # 2.81.1
+      fetchSubmodules= false;
+    };
+
+    nativeBuildInputs = [ cmake fakeGit git makeWrapper unzip cjson ];
+    buildInputs = [
+      glew SDL2 zlib minizip libjpeg curl lua libogg libtheora freetype libpng sqlite openal
+    ];
+
+    preBuild = ''
+      export SOURCE_DATE_EPOCH=$(date +%s)
+      export CI="true"
+    '';
+
+    cmakeFlags = [
+      "-DCMAKE_BUILD_TYPE=Release"
+      "-DCROSS_COMPILE32=0"
+      "-DBUILD_SERVER=0"
+      "-DBUILD_CLIENT=1"
+      "-DBUNDLED_JPEG=0"
+      "-DBUNDLED_LIBS=0"
+      "-DINSTALL_EXTRA=0"
+      "-DINSTALL_OMNIBOT=0"
+      "-DINSTALL_GEOIP=0"
+      "-DINSTALL_WOLFADMIN=0"
+      "-DFEATURE_AUTOUPDATE=0"
+      "-DINSTALL_DEFAULT_BASEDIR=."
+      "-DINSTALL_DEFAULT_BINDIR=."
+      "-DINSTALL_DEFAULT_MODDIR=."
+    ];
+
+    postInstall = ''
+      ETMAIN=$out/etmain
+      mkdir -p $ETMAIN
+      ln -s ${pak0} $ETMAIN/pak0.pk3
+      ln -s ${pak1} $ETMAIN/pak1.pk3
+      ln -s ${pak2} $ETMAIN/pak2.pk3
+    '';
+
+    meta = with lib; {
+      description = "ET: Legacy is an open source project based on the code of Wolfenstein: Enemy Territory which was released in 2010 under the terms of the GPLv3 license";
+      homepage = "https://etlegacy.com";
+      platforms = [ "i686-linux" "x86_64-linux" ];
+      license = licenses.gpl3;
+      maintainers = with maintainers; [ ashleyghooper ];
+    };
+  };
+in runCommand pkgname { buildInputs = [ makeWrapper ]; } ''
+  BIN=$out/bin
```

Comment:
why not merge this in the "gamedir" derivation ? this would expose gamedir derivation so that users can override it, which is currently not possible

Reply:
@teto, just to clarify, is it the `runCommand` script you're suggesting to move up to the `gamedir` section?

Reply:
exactly. the runCommand script could go into the postInstall phase, couldn't it ? there are examples in nixpkgs, just grep for it

Reply:
Thanks for clarifying. This game is cross platform and expects to find the resource files to be located with the game binary.

The directory structure once built looks like below.

```
/nix/store/w2wi7l2apzmq28501z290gvzmda96va4-etlegacy-2.81.1
├── etl.i386
├── etmain
│   ├── campaigncycle.cfg
│   ├── etl_server.cfg
│   ├── etl_server_comp.cfg
│   ├── legacy.cfg
│   ├── lmscycle.cfg
│   ├── mapvotecycle.cfg
│   ├── mapvoteplayerscount.cfg
│   ├── objectivecycle.cfg
│   ├── pak0.pk3 -> /nix/store/595dvac3wlnwql05layq72fhi0wpzraz-pak0.pk3
│   ├── pak1.pk3 -> /nix/store/5dhgs44wjjz2r175ksavbhzmrjw1r1qp-pak1.pk3
│   └── pak2.pk3 -> /nix/store/yszmhgv605chj3dn9flazcaggyfrwsb5-pak2.pk3
├── legacy
│   ├── description.txt
│   ├── legacy_2.81.1.pk3
│   └── qagame.mp.i386.so
├── librenderer_opengl1_i386.so
```

Having the `runCommand` as output is so that we can create a wrapper script that changes directory to where the `gamedir` abstraction that produces the above is located in the nix store before calling the main binary (`etl.i386` or `etl.x86_64`), otherwise the game fails to start because it is unable to find the resource files. And it also propagates the XDG `.desktop` files into `share/` and gets the main binary into `$PATH` so that they can find it.

I think moving the `runCommand` logic up into the `postInstall` phase would create a circular dependency, because we won't know the output of `gamedir` until `gamedir` is built, specifically here:

```
makeWrapper ${gamedir}/$EXE $BIN/$EXE --chdir ${gamedir}
```

In light of the above, is there a better way we could be approaching this though?

Reply:
Hi @teto, did my reply above make sense? Is there another approach I could be taking instead given the project layout?

Reply:
Hi @teto, with some help from the folks on discord I was able to refactor the package to use a single `stdenv.mkderivation`. I have now pushed the updated version to my fork so it's active for this PR.

Reply:
sry for lack of follow up. Looks good !